test/test_genome.py

```diff
@@ -346,89 +347,118 @@ def test_is_gene_in_output_region(rng_seed):
-    assert genome._mutate_hidden_region(list(dna), 8, active_regions, rng) is True
 
-    # mutating any gene in active region without silent genes returns False
-    assert genome._mutate_hidden_region(list(dna), 9, active_regions, rng) is False
-    assert genome._mutate_hidden_region(list(dna), 10, active_regions, rng) is False
-    assert genome._mutate_hidden_region(list(dna), 11, active_regions, rng) is False
+def test_only_silent_mutations(genome_params, rng_seed):
+    genome = cgp.Genome(**genome_params)
+    rng = np.random.RandomState(rng_seed)
+    genome.randomize(rng)
 
+    only_silent_mutations_0 = genome.mutate(mutation_rate=0, rng=rng)
+    assert only_silent_mutations_0
 
-def test_mutate_output_region(rng_seed):
-    rng = np.random.RandomState(rng_seed)
-    genome = cgp.Genome(1, 1, 2, 1, None, (cgp.Add,))
-    dna = [
-        ID_INPUT_NODE,
-        ID_NON_CODING_GENE,
-        ID_NON_CODING_GENE,
-        0,
-        0,
-        0,
-        0,
-        0,
-        0,
-        ID_OUTPUT_NODE,
+    only_silent_mutations_1 = genome.mutate(mutation_rate=1, rng=rng)
+    assert not only_silent_mutations_1
+
+
+# Overwrite global genome_params() from conftest,
+# since tests below strongly depend on these variable values
+@fixture
+def genome_params():
+    return {
+        "n_inputs": 2,
+        "n_outputs": 1,
+        "n_columns": 3,
+        "n_rows": 3,
+        "levels_back": 2,
+        "primitives": (cgp.Add, cgp.Sub, cgp.ConstantFloat),
+    }
+
+
+def test_permissible_values_hidden(genome_params, rng_seed):
+    genome = cgp.Genome(**genome_params)
+
+    # test function gene
+    gene_idx = 6  # function gene of first hidden gene
+    gene = 0  #
+    region_idx = 1000  # can be any number, since not touched for function gene
+    permissible_function_gene_values = genome._determine_permissible_values_hidden(
+        gene_idx, gene, region_idx
+    )
+
+    # test input gene
+    gene_idx = 16  # first input gene in second hidden layer
+    gene = 0
+    region_idx = 5
+    permissible_hidden_input_gene_values = genome._determine_permissible_values_hidden(
+        gene_idx, gene, region_idx
+    )
+
+    assert permissible_function_gene_values == [
+        1,
         2,
-        ID_NON_CODING_GENE,
-    ]
+    ]  # function idx 1,2 (cgp.Sub, cgp.ConstantFloat)
+    assert permissible_hidden_input_gene_values == [
+        1,
+        2,
+        3,
+        4,
+    ]  # gene indices of input, 1st hidden layer
 
-    assert genome._mutate_output_region(list(dna), 9, rng) is True
-    assert genome._mutate_output_region(list(dna), 10, rng) is False
-    assert genome._mutate_output_region(list(dna), 11, rng) is True
 
+def test_permissible_values_output(genome_params, rng_seed):
+    genome = cgp.Genome(**genome_params)
 
-@pytest.mark.parametrize("mutation_rate", [0.02, 0.05, 0.2])
-def test_correct_number_of_mutations(mutation_rate, rng_seed):
+    gene_idx_0 = 33
+    gene_idx_1 = 34
+    gene_idx_2 = 35
```

Comment:
```suggestion
    gene_idx_function = 33
    gene_idx_input0 = 34
    gene_idx_input1 = 35
```
just to make this more easily understandable